.github/workflows/cibuildwheel.yml

```diff
@@ -90,35 +90,21 @@ jobs:
     name: Build wheels on ${{ matrix.os }} for ${{ matrix.cibw_archs }}
     runs-on: ${{ matrix.os }}
     env:
-      # The following commands branch based on whether we are on the main
-      # branch or have a PR into main.  For these, we use the NumPy 2.0
-      # nightlies to ensure future C-API/ABI compatibility.
-      # This branching becomes unnecessary when NumPy 2.0 is released.
-      # When using no-build-isolation we need to install all requirements.
-      CIBW_BEFORE_BUILD:
-        ${{ (((github.event_name == 'push' && github.ref == 'refs/heads/main') ||
-              (github.event_name == 'pull_request' && github.base_ref == 'refs/heads/main')) &&
-            'pip install meson-python pybind11 setuptools_scm "setuptools>=64" ninja &&
-             pip install "numpy>=2.0.0.dev0" --extra-index-url "https://pypi.anaconda.org/scientific-python-nightly-wheels/simple" &&
-             rm -rf {package}/build'
-          ) ||
-            'pip install "numpy>=1.25" &&
-             rm -rf {package}/build'
-        }}
-      CIBW_BEFORE_BUILD_WINDOWS:
-        ${{ (((github.event_name == 'push' && github.ref == 'refs/heads/main') ||
-              (github.event_name == 'pull_request' && github.base_ref == 'refs/heads/main')) &&
-            'pip install delvewheel meson-python pybind11 setuptools_scm "setuptools>=64" ninja &&
-             pip install "numpy>=2.0.0.dev0" --extra-index-url "https://pypi.anaconda.org/scientific-python-nightly-wheels/simple" &&
-             rm -rf {package}/build'
-          ) ||
-            'pip install delvewheel "numpy>=1.25" &&
-             rm -rf {package}/build'
-        }}
+      CIBW_BEFORE_BUILD: >-
+        rm -rf {package}/build
+      CIBW_BEFORE_BUILD_WINDOWS: >-
+        pip install delvewheel &&
+        rm -rf {package}/build
+      # Live on the edge when building on main or a PR against main since
+      # these wheels are also used for nightlies and NumPy 2.0 transition
+      # requires using the NumPy 2.0 nightlies for compatibility.
+      # If using all `--pre` releases creates issues, the NumPy wheel can be
+      # istalled more targeted.
       CIBW_BUILD_FRONTEND: >-
-        ${{ (((github.event_name == 'push' && github.ref == 'refs/heads/main') ||
+        ${{ (((github.event_name == 'main' && github.ref == 'refs/heads/main') ||
```

Comment:
```suggestion
        ${{ (((github.event_name == 'push' && github.ref == 'refs/heads/main') ||
```

Reply:
Uggg, sorry about messing it up twice :/